contracts/test/EscrowEchidnaTest.sol

```diff
@@ -0,0 +1,79 @@
+
+    function deploy() external returns (address addr) {
+        bytes memory bytecode = daiBytecode;
+        assembly {
+            addr := create(0, add(bytecode, 0x20), mload(bytecode))
+        }
+    }
+}
+
+/// @dev Escrow Echidna Testing
+contract EscrowEchidnaTest {
+
+    address internal dai;
+    L1Escrow internal escrow;
+    MockMessenger internal messenger1;
+    MockMessenger internal messenger2;
+    L2Gateway internal gate2;
+    L1Gateway internal gate1;
+    Dai internal oDai;
+    Create internal create;
+
+    uint256 internal constant WAD = 10**18;
+    uint256 internal constant MAX_SUPPLY = 10**15 * WAD;
+
+    /// @dev
+    constructor () {
+        create = new Create();
+        dai = create.deploy();
+        oDai = new Dai();
+        escrow = new L1Escrow();
+        messenger1 = new MockMessenger();
+        messenger2 = new MockMessenger();
+        gate2 = new L2Gateway(address(messenger2), address(oDai));
+        gate1 = new L1Gateway(TokenLike(dai), address(gate2), address(messenger1), address(escrow));
+    }
+
+    // --- Math ---
+    function add(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x + y;
+        assert (z >= x); // check if there is an addition overflow
+    }
+    function sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x - y;
+        assert (z <= x); // check if there is a subtraction overflow
+    }
+
+    /// @dev
+    function bridge(uint256 wad) public {
+        uint256 supply = DaiLike(dai).totalSupply();
+        uint256 holderBalance = DaiLike(dai).balanceOf(address(this));
+        L1Escrow(escrow).approve(dai, address(this), type(uint256).max);
```

Comment:
This line could be moved to the constructor, I believe?

Also, doesn't the escrow contract need to approve the L1Gateway, not the fuzzing contract?
https://github.com/makerdao/optimism-dai-bridge/blob/master/contracts/l1/L1Gateway.sol#L79

Reply:
Done.